third-party/realdds/src/dds-device-broadcaster.cpp

```diff
@@ -4,45 +4,45 @@
 #include <realdds/dds-device-broadcaster.h>
 
 #include <realdds/dds-participant.h>
+#include <realdds/dds-publisher.h>
+#include <realdds/dds-topic-writer.h>
 #include <realdds/dds-utilities.h>
 #include <realdds/dds-guid.h>
-#include <realdds/topics/device-info/device-info-msg.h>
-#include <realdds/topics/device-info/deviceInfoPubSubTypes.h>
+#include <realdds/topics/flexible/flexible-msg.h>
+#include <realdds/topics/flexible/flexiblePubSubTypes.h>
 
 #include <librealsense2/utilities/easylogging/easyloggingpp.h>
-#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
-#include <fastdds/dds/domain/DomainParticipant.hpp>
-#include <fastdds/dds/domain/DomainParticipantListener.hpp>
-#include <fastdds/dds/publisher/Publisher.hpp>
-#include <fastdds/dds/publisher/DataWriter.hpp>
-#include <fastdds/dds/topic/TypeSupport.hpp>
-#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
-#include <fastdds/rtps/participant/ParticipantDiscoveryInfo.h>
+#include <librealsense2/utilities/json.h>
 
 #include <iostream>
 
 using namespace eprosima::fastdds::dds;
 using namespace realdds;
 
 // We want to know when readers join our topic
-class dds_device_broadcaster::dds_client_listener : public eprosima::fastdds::dds::DataWriterListener
+class dds_device_broadcaster::dds_client_listener : public realdds::dds_topic_writer //dds_topic_writer is a listener
 {
 public:
-    dds_client_listener( dds_device_broadcaster * owner )
-        : eprosima::fastdds::dds::DataWriterListener()
+    dds_client_listener( std::shared_ptr< dds_topic > const & topic,
+                         std::shared_ptr< dds_publisher > const & publisher,
+                         dds_device_broadcaster * owner )
+        : dds_topic_writer( topic, publisher )
         , _owner( owner )
     {
     }
 
+    std::atomic_bool _new_reader_joined = { false };  // Used to indicate that a new reader has joined for this writer
+
+protected:
     void on_publication_matched( eprosima::fastdds::dds::DataWriter * writer,
-                                 const eprosima::fastdds::dds::PublicationMatchedStatus & info ) override
+                                 eprosima::fastdds::dds::PublicationMatchedStatus const & info ) override
```

Comment:
General question , this is the same right?
const c &
const & c

Reply:
`const T &` and `T const &` are the same.
I changed to match the definition in `dds_topic_writer` when something didn't work and I couldn't understand why. This was not the reason but I left it this way :-)